include/circt/Dialect/FIRRTL/Dialect.h

```diff
@@ -29,6 +29,9 @@ class FIRRTLDialect : public Dialect {
   static StringRef getDialectNamespace() { return "firrtl"; }
 };
 
+/// The string used to name the FIRRTL name attribute
+const StringRef kFIRRTLName = "firrtl.name";
```

Comment:
i'm not sure the policies or best practices on constants, but i saw this in a few places so i thought i'd make a constant for it.